modules/crawler/src/main/scala/Downloader.scala

```diff
@@ -0,0 +1,92 @@
+package fide.crawler
+
+import cats.effect.IO
+import cats.syntax.all.*
+import fide.domain.*
+import fide.types.*
+import org.http4s.*
+import org.http4s.client.Client
+import org.http4s.implicits.*
+import org.typelevel.log4cats.Logger
+import org.typelevel.log4cats.syntax.*
+
+type PlayerInfo = (NewPlayer, Option[NewFederation])
+trait Downloader:
+  def fetch: fs2.Stream[IO, PlayerInfo]
+
+object Downloader:
+  val downloadUrl = uri"http://ratings.fide.com/download/players_list.zip"
+
+  lazy val request = Request[IO](
+    method = Method.GET,
+    uri = downloadUrl
+  )
+
+  def apply(client: Client[IO])(using Logger[IO]): Downloader = new:
+
+    def fetch =
+      client
+        .stream(request)
+        .switchMap(_.body)
+        .through(Decompressor.decompress)
+        .through(fs2.text.utf8.decode)
+        .through(fs2.text.lines)
+        .drop(1) // first line is header
+        .evalMap(parseLine)
+        .collect { case Some(x) => x }
```

Comment:
We can shorten it to just `.evalMapFilter(parseLine)` ([source](https://github.com/typelevel/fs2/blob/8a67a3b4899ee43a9efa642e5f362d2f96be5dbd/core/shared/src/main/scala/fs2/Stream.scala#L1112-L1116))

Reply:
fixed https://github.com/lenguyenthanh/fide/pull/268/files